### DISCUSSION TITLE
How to store sensitive data for configuration rendering in Netbox

### DISCUSSION THREAD

**Discussion by whtkam** [netbox-community/netbox#15392] [Q&A]
I want to store the device configuration passwords in the Netbox Secrets plugin. The idea is that when rendering the configuration in Netbox I specify the session key in advance. So the configuration is generated with all passwords. If there is no session key, no password information is included in the rendered configurations.
The problem I'm having is that I don't know how to access the netbox-secrets plugin's plaintext password in a Jinja2 template.
I have no idea where to look - the hierarchy in the Swaager UI doesn't work.
There may be a better solution to store the sensitive data for the device configurations.
Thanks

**Comment by PaulR282** [comment]
I had the same problem and the same idea. AFAIK you can't query data in jinja templates in netbox. I haven't found a good solution for storing secrets and using them in plaintext in a config template.

**Comment by irrwitzer42** [comment]
On my journey I sumbled upon this posting. Afterwards I found this one: https://github.com/Onemind-Services-LLC/netbox-secrets/discussions/102 haven't tried it yet, but looks promising and sounds reasonable.

**Reply by jhofmueller** [reply]
Actually we do use this plugin for exactly the tasks mentioned above. Everything ist peachy except for the very loooooong time it takes the maintainers to release new versions when a new major NetBox version is released.
